tests/testthat/test-checking.R

```diff
@@ -0,0 +1,33 @@
+test_that("check_dist_func works as expected", {
+  expect_true(check_dist_func(func = \(n) rgamma(n = n, shape = 2, scale = 2)))
+})
+
+test_that("check_dist_func errors when not a function", {
+  expect_error(
+    check_dist_func(func = "function"),
+    regexp = "(Assertion)*(func)*(failed)*(Must be a function, not)*(character)"
```

Comment:
I'm a bit wary of so-prescriptive checking of error messages; if we *want* to guarantee that error messages contain certain kinds of helpful information (e.g. class of invalid argument), I'm game for that - but that means testing n>1 alternatives.

But other structural bits, like the "Assertion" match - meh, doesn't seem necessary.